pkgs/by-name/cl/cljstyle/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  buildGraalvmNativeImage,
+  fetchMavenArtifact,
+  fetchurl,
+  graalvmPackages,
+}:
+
+let
+  # must be on classpath to build native image
+  graal-build-time = fetchMavenArtifact {
+    repos = [ "https://repo.clojars.org/" ];
+    groupId = "com.github.clj-easy";
+    artifactId = "graal-build-time";
+    version = "1.0.5";
+    sha256 = "sha256-M6/U27a5n/QGuUzGmo8KphVnNa2K+LFajP5coZiFXoY=";
+  };
+in
+buildGraalvmNativeImage rec {
+  pname = "cljstyle";
+  version = "0.17.642";
+
+  src = fetchurl {
+    url = "https://github.com/greglook/${pname}/releases/download/${version}/${pname}-${version}.jar";
+    sha256 = "sha256-AkCuTZeDXbNBuwPZEMhYGF/oOGIKq5zVDwL8xwnj+mE=";
+  };
+
+  graalvmDrv = graalvmPackages.graalvm-ce;
+
+  extraNativeImageBuildArgs = [
+    "-H:+ReportExceptionStackTraces"
+    "--no-fallback"
+    "-cp ${graal-build-time.passthru.jar}"
+  ];
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    $out/bin/cljstyle --help
+    $out/bin/cljstyle version | grep "cljstyle ${version}"
+
+    runHook postInstallCheck
+  '';
+
+  meta = with lib; {
+    description = "A tool for formatting Clojure code";
+    homepage = "https://github.com/greglook/cljstyle";
+    sourceProvenance = [ sourceTypes.binaryBytecode ];
+    license = licenses.epl10;
+    changelog = "https://github.com/greglook/cljstyle/blob/${version}/CHANGELOG.md";
+    maintainers = [ maintainers.psyclyx ];
+  };
```

```suggestion
  mainProgram = "cljstyle";
  };
```
Since `bin/cljstyle` exists